hello-app-iap/main.go

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START all]
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+func main() {
+	// use port1 serves service1 and port2 serves service2
+	const (
+		port1 = "8081"
+        	port2 = "8082"
+	)
+
+	finish := make(chan bool)
+
+	server1 := http.NewServeMux()
+	server1.HandleFunc("/", hello1)
+
+    	server2 := http.NewServeMux()
+	server2.HandleFunc("/", hello2)
+
+   	go func() {
+        	log.Printf("Server listening on port %s", port1)
+	    	err := http.ListenAndServe(":"+port1, server1)
+        	log.Fatal(err)
+	}()
+
+	go func() {
+        	log.Printf("Server listening on port %s", port2)
+	    	err := http.ListenAndServe(":"+port2, server2)
+        	log.Fatal(err)
+	}()
+
+    	<-finish
+	
+}
+
+// hello1 prints "Hello world from service1"
+func hello1(w http.ResponseWriter, r *http.Request) {
```

Comment:
hmm you can potentially merge hello1 hello2 into one method and just log:


`log.Printf("received request on address %s", req.Host)`

would that work?

Reply:
Yes it works. But I want to make this demo capable of multiple domains in the future. Or it can work for multiple domains with minor changes. So I am still prone to keep two ports here. Make sense?

Reply:
I guess you can keep separate, it's ok.